src/ucp/wireup/wireup.c

```diff
@@ -210,9 +215,178 @@ static void ucp_wireup_msg_get_peer_name(const ucp_wireup_msg_t *msg,
     peer_name[length] = '\0';
 }
 
-static struct sockaddr * ucp_wireup_msg_get_aux_addr(const ucp_wireup_msg_t *msg)
+static void ucp_wireup_msg_get_tl_name(const ucp_wireup_msg_t *msg,
+                                       char *tl_name, size_t max)
 {
-    return (struct sockaddr *)((void*)(msg + 1) + msg->addr_len);
+    size_t length = ucs_min(max - 1, msg->tl_name_len);
+    memcpy(tl_name, (const char*)(msg + 1) + msg->peer_name_len, length);
+    tl_name[length] = '\0';
+}
+
+static struct sockaddr * ucp_wireup_msg_get_aux_addr(ucp_wireup_msg_t *msg)
+{
+    void *msg_data = msg + 1;
+    return msg_data + msg->peer_name_len + msg->tl_name_len + msg->addr_len;
+}
+
+static double ucp_aux_score_func(ucp_worker_h worker, uct_iface_attr_t *iface_attr)
+{
+    if (!(iface_attr->cap.flags & UCT_IFACE_FLAG_AM_BCOPY) || /* Need to use it for wireup messages */
+        !(iface_attr->cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) || /* Should connect immediately */
+        !(iface_attr->cap.flags & UCT_IFACE_FLAG_AM_CB_ASYNC) || /* Should progress asynchronously */
```

Comment:
this is needed to enable establishing connections using an auxiliary transport without explicitly calling the progress function of the auxiliary transport. For example, to avoid deadlock with user code, which may be calling something like PMI barrier.